### PR TITLE
add filter for blacklist/whitelist.

### LINKED ISSUE #14: Add white/black list
**What would you like to be added**:
Filter for white/black, can be ip address, backend endpoint.

**Why is this needed**:
Control if ip can request to backend or backend can be call by proxy

### PR DESCRIPTION
**Pull request by oaoit**
**What this PR does**:
add filter for http blacklist/whitelist.
**Which issue(s) this PR fixes**:

Fixes #14 

**Special notes for your reviewer**:
When IP address or backend application name are on the blacklist, the call will stop, response 403.
When IP address or backend application name are not on the whitelist, the call will stop, response 403.
**Does this PR introduce a user-facing change?**:
There is no.

```release-note

```

### CODE REVIEW

File: pkg/context/http/context.go [hunk 1]
```diff
@@ -115,6 +118,33 @@ func (hc *HttpContext) GetApi() *model.Api {
 	return hc.api
 }
 
+func (hc *HttpContext) GetClientIP() string {
+	xForwardedFor := hc.Request.Header.Get("X-Forwarded-For")
+	ip := strings.TrimSpace(strings.Split(xForwardedFor, ",")[0])
+	if ip != "" {
```
Comment (williamfeng323) [inline pkg/context/http/context.go:124]: please use len(ip) == 0

Comment (williamfeng323) [reply pkg/context/http/context.go:124]: same as below

File: pkg/filter/authority_filter.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+package filter
+
+import (
+	"github.com/dubbogo/dubbo-go-proxy/pkg/common/constant"
+	"github.com/dubbogo/dubbo-go-proxy/pkg/common/extension"
+	"github.com/dubbogo/dubbo-go-proxy/pkg/context"
+	"github.com/dubbogo/dubbo-go-proxy/pkg/context/http"
+	"github.com/dubbogo/dubbo-go-proxy/pkg/model"
+	nh "net/http"
```
Comment (AlexStocks) [inline pkg/filter/authority_filter.go:26]: split the import block and move this package to the 1st import block.

**Review by cityiron — APPROVED** [review]

**Review by williamfeng323 — APPROVED** [review]
lgtm

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**